src/GroupManager.php

```diff
@@ -121,13 +137,59 @@ public function removeGroup($entity_type_id, $bundle_id) {
       $editable->set('groups', $groups);
       $saved = $editable->save();
 
+      // Remove all roles associated with this group type.
+      $this->removeRoles($entity_type_id, $bundle_id);
+
       $this->refreshGroupMap();
 
       return $saved;
     }
   }
 
   /**
+   * Creates default roles for the given group type.
+   *
+   * @param string $entity_type_id
+   *   The entity type ID of the group for which to create default roles.
+   * @param string $bundle_id
+   *   The bundle ID of the group for which to create default roles.
+   */
+  protected function createDefaultRoles($entity_type_id, $bundle_id) {
+    $properties = [
+      'group_type' => $entity_type_id,
+      'group_bundle' => $bundle_id,
+    ];
+    foreach ([OgRoleInterface::ANONYMOUS, OgRoleInterface::AUTHENTICATED, OgRoleInterface::ADMINISTRATOR] as $role_name) {
+      // @todo: Find a way to deal with potential ID collisions.
+      $properties['id'] = $role_name;
+      $properties['role_type'] = OgRole::getRoleTypeByName($role_name);
+      // Only create the default role if it doesn't exist yet.
+      if (!$this->ogRoleStorage->loadByProperties($properties)) {
```

Comment:
I think We can make sure the sql schema will.prevent a duplicate (if that's not already the case), thus avoid babysitting the code.